WalletWasabi.Fluent/Controls/ResponsivePanel.cs

```diff
@@ -0,0 +1,219 @@
+        {
+            Contract.Requires<ArgumentNullException>(element != null);
+            element!.SetValue(RowSpanProperty, value);
+        }
+
+        public double ItemWidth
+        {
+            get => GetValue(ItemWidthProperty);
+            set => SetValue(ItemWidthProperty, value);
+        }
+
+        public double ItemHeight
+        {
+	        get => GetValue(ItemHeightProperty);
+	        set => SetValue(ItemHeightProperty, value);
+        }
+
+        public double AspectRatio
+        {
+	        get => GetValue(AspectRatioProperty);
+	        set => SetValue(AspectRatioProperty, value);
+        }
+
+        public AvaloniaList<int> ColumnHints
+        {
+            get => GetValue(ColumnHintsProperty);
+            set => SetValue(ColumnHintsProperty, value);
+        }
+
+        public AvaloniaList<double> WidthTriggers
+        {
+            get => GetValue(WidthTriggersProperty);
+            set => SetValue(WidthTriggersProperty, value);
+        }
+
+        static ResponsivePanel()
+        {
+            AffectsParentMeasure<ResponsivePanel>(AspectRatioProperty, ColumnHintsProperty, WidthTriggersProperty, ColumnSpanProperty, RowSpanProperty);
+            AffectsParentArrange<ResponsivePanel>(AspectRatioProperty, ColumnHintsProperty, WidthTriggersProperty, ColumnSpanProperty, RowSpanProperty);
+            AffectsMeasure<ResponsivePanel>(AspectRatioProperty, ColumnHintsProperty, WidthTriggersProperty, ColumnSpanProperty, RowSpanProperty);
+            AffectsArrange<ResponsivePanel>(AspectRatioProperty, ColumnHintsProperty, WidthTriggersProperty, ColumnSpanProperty, RowSpanProperty);
+        }
+
+        private struct Item
+        {
+            public int Column;
+            public int Row;
+            public int ColumnSpan;
+            public int RowSpan;
+        }
+
+        private Size MeasureArrange(Size panelSize, bool isMeasure)
+        {
+            var children = Children;
+            var widthTriggers = WidthTriggers;
+            var columnHints = ColumnHints;
+            var aspectRatio = AspectRatio;
+            var width = panelSize.Width;
+            var height = panelSize.Height;
+
+            if (widthTriggers.Count <= 0)
+            {
+                throw new Exception($"No width trigger specified in {nameof(WidthTriggers)} property.");
+            }
+
+            if (columnHints.Count <= 0)
+            {
+                throw new Exception($"No column hints specified in {nameof(ColumnHints)} property.");
+            }
+
+            if (widthTriggers.Count != columnHints.Count)
+            {
+                throw new Exception($"Number of width triggers must be equal to the number of column triggers.");
+            }
+
+            if (double.IsNaN(aspectRatio))
+            {
+                if (height == 0 || double.IsInfinity(height))
+                {
+                    aspectRatio = 1.0;
+                }
+                // else
```

Comment:
@wieslawsoltes is this just some leftover code or is it needed in the future?

Reply:
This is still under active development so for now I would leave this code in.